archivebox/cli/archivebox_add.py

```diff
@@ -89,7 +89,7 @@ def main(args: Optional[List[str]]=None, stdin: Optional[IO]=None, pwd: Optional
     parser.add_argument(
         "--parser",
         type=str,
-        help="Parser used to read inputted URLs.",
+        help="Parser used for input on stdin (not valid with URLs).",
```

Comment:
To clarify, parser still applies to URLs passed as CLI args in some cases. The behavior is subtle and maybe a bit too "magic", but the intent is to work for cases like this:

```bash
archivebox add --depth=0 --parser=rss 'https://dropbox.tech/feed'
```

Without needing the user to do this:
```bash
curl 'https://dropbox.tech/feed' | archivebox add --depth=1 --parser=rss
# or
archivebox add --depth=1 --parser=rss 'https://dropbox.tech/feed'
```

See this this case in particular here:
`--depth=0 + --parser=rss + passed URL(s) or local file path(s) via stdin/args`
https://github.com/ArchiveBox/ArchiveBox/issues/1363#issuecomment-1966177173

Honestly I'm still conflicted about the level of magic needed to get my original intent here working, as it introduces a ton of edge cases / unexpected results around `--depth=0` and `--depth=1` interacting with `--parser=`.

I could be convinced that we should remove all of my "guess what the user was trying to do" magic entirely and just better document which CLI args/stdin/etc. need to be used for each case.

Reply:
Okay, I think I understand better now how it was supposed to work. The intersection of parser bugs I was hitting along with the most-links behavior of auto was making me more confused than I should have been. I’ll dig into it more soon, maybe tomorrow.